src/commands/integration/deploy.ts

```diff
@@ -394,8 +394,7 @@ export const getConfiguration = (workingDir) => {
 export const deploy = async (options: OptionValues, command: BaseCommand) => {
   const { api, cachedConfig, site, siteInfo } = command.netlify
   const { id: siteId } = site
-  // @ts-expect-error TS(2554) FIXME: Expected 1 arguments, but got 0.
-  const [token] = await getToken()
+  const [token] = await command.authenticate(options.auth)
```

Comment:
Note that this one will have a potential change in behavior - by using `command.authenticate` (like all the other commands) - this will start auth flow if user is not logged in (or didn't provide token via env var or `--auth`), while previously it would result in getting no token (and presumably getting failures later down the road?)

After fixing up bunch of related TypeScript elsewhere, using the `getToken` that could be `null` (previously it would be just `any`) it would complain about `token` being possibly `null` in https://github.com/netlify/cli/blob/e64e30c66d4612836926d27d23fe12e97726582e/src/commands/integration/deploy.ts#L421-L428

so I assume this should be required that token is present before executing the rest of this command?

cc @netlify/ecosystem-pod-composable-tooling

Reply:
This is probably wrong thing to do, as we do authenticate in BaseCommand already - I will be looking if maybe there is some nice way to avoid token being typed as potentially `null`

Reply:
Ops, my bad - we don't actually force authenticate in base command, so this change seems still good here